storage/google/cloud/storage/blob.py

```diff
@@ -404,14 +408,22 @@ def _get_download_url(self):
         :returns: The download URL for the current blob.
         """
         if self.media_link is None:
-            download_url = _DOWNLOAD_URL_TEMPLATE.format(path=self.path)
+            base_url = _DOWNLOAD_URL_TEMPLATE.format(path=self.path)
+            scheme, netloc, path, query, frag = urlsplit(base_url)
+            query = parse_qsl(query)
+            # Only add `generation' for synthesized URLs: 'mediaLink' from
+            # server presumably already encodes it.
```

Comment:
Nope:  I'm explaining why I am adding `generation` in the branch which *doesn't* have `mediaUrl` set by the server.